gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiSubscriptionsResource_RenewApiKeysTest.java

```diff
@@ -156,6 +157,28 @@ public void should_return_400_if_application_is_in_shared_api_key_mode() {
         assertEquals("Invalid operation for API Key mode [SHARED] of application [my-application].", error.getMessage());
     }
 
+    @Test
+    public void should_return_400_if_subscription_is_closed() {
+        final SubscriptionEntity subscriptionEntity = SubscriptionFixtures
+            .aSubscriptionEntity()
+            .toBuilder()
+            .id(SUBSCRIPTION)
+            .api("ANOTHER-API")
+            .status(SubscriptionStatus.CLOSED)
+            .build();
+
+        when(subscriptionService.findById(SUBSCRIPTION)).thenReturn(subscriptionEntity);
+
+        final Response response = rootTarget().request().post(Entity.json(SubscriptionFixtures.aRenewApiKey()));
+        assertEquals(BAD_REQUEST_400, response.getStatus());
+
+        var error = response.readEntity(Error.class);
+        assertEquals(BAD_REQUEST_400, (int) error.getHttpStatus());
+        assertEquals("Subscription [my-subscription] should be paused or accepted. Currently it is CLOSED", error.getMessage());
+
+        verify(subscriptionService, never()).pause(any(), any());
+    }
+
```

Comment:
If you want to keep this test, you will have to mock apikeyservice so it returns an exception.